lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+                .unwrap_or(destination_component_id);
+
+            if apa.targets().map_or(true, |targets| {
+                targets.destination_component_id == destination_component_id
+            }) {
+                // If the "source" Component is marked for deletion, and we (the destination) are
+                // *NOT*, then we should ignore the argument as data should not flow from things
+                // that are marked for deletion to ones that are not.
+                let destination_component = Component::get_by_id(ctx, destination_component_id)
+                    .await
+                    .map_err(|e| AttributeValueError::Component(e.to_string()))?;
+
+                let source_component = Component::get_by_id(ctx, expected_source_component_id)
+                    .await
+                    .map_err(|e| AttributeValueError::Component(e.to_string()))?;
+
+                if source_component.to_delete() && !destination_component.to_delete() {
+                    continue;
+                }
+
+                let func_arg_id =
+                    AttributePrototypeArgument::func_argument_id_by_id(ctx, apa_id).await?;
+                let func_arg_name = ctx
+                    .workspace_snapshot()?
+                    .get_node_weight_by_id(func_arg_id)
+                    .await?
+                    .get_func_argument_node_weight()?
+                    .name()
+                    .to_owned();
+
+                let values_for_arg =
+                    match AttributePrototypeArgument::value_source_by_id(ctx, apa_id)
+                        .await?
+                        .ok_or(
+                            AttributeValueError::AttributePrototypeArgumentMissingValueSource(
+                                apa_id,
+                            ),
+                        )? {
+                        ValueSource::StaticArgumentValue(static_argument_value_id) => {
+                            let val = StaticArgumentValue::get_by_id(ctx, static_argument_value_id)
+                                .await?
+                                .value;
+                            arg_sources.insert(
+                                func_arg_name.clone(),
+                                Some(PropPath::new(["Static Value"]).to_string()),
+                            );
+                            vec![val]
+                        }
+                        other_source => {
+                            let mut values = vec![];
+
+                            for av_id in other_source
+                                .attribute_values_for_component_id(
+                                    ctx,
+                                    expected_source_component_id,
+                                )
+                                .await?
+                            {
+                                let attribute_value = AttributeValue::get_by_id(ctx, av_id).await?;
```

Comment:
A smell for me when match arms get too big is when you have a scope that contains non-trivial iterator or recursive logic in them. I'd look into breaking this up.

Example:
```rust
match foo {
    Some(v) => println!("{v}"),
    None => {
        for i in big_ass_vec {
            // a bunch of code
        }
```